lib/OpenQA/Worker/Cache/Service.pm

```diff
@@ -0,0 +1,206 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Service;
+
+use Mojolicious::Lite;
+use OpenQA::Worker::Cache::Task::Asset;
+use Mojolicious::Plugin::Minion;
+use OpenQA::Worker::Cache qw(STATUS_PROCESSED STATUS_ENQUEUED STATUS_DOWNLOADING STATUS_IGNORE STATUS_ERROR);
+use Mojo::Collection;
+use Mojolicious::Plugin::Minion::Admin;
+
+use constant DEFAULT_MINION_WORKERS => 5;
+
+BEGIN { srand(time) }
+
+my $_token;
+
+sub _gen_session_token { $_token = int(rand(999999999999)) }
+
+my $enqueued = Mojo::Collection->new;
+app->hook(
+    before_server_start => sub {
+        _gen_session_token();
+        $enqueued = Mojo::Collection->new;
+    });
+
+plugin Minion => {SQLite => 'sqlite:' . OpenQA::Worker::Cache->from_worker->db_file};
+plugin 'Minion::Admin';
+plugin 'OpenQA::Worker::Cache::Task::Asset';
+plugin 'OpenQA::Worker::Cache::Task::Sync';
+
+sub SESSION_TOKEN { $_token }
+
+sub _gen_guard_name { join('.', SESSION_TOKEN, pop) }
+
+sub _exists { !!(defined $_[0] && exists $_[0]->{total} && $_[0]->{total} > 0) }
+
+sub active { !app->minion->lock(shift, 0) }
+
+sub enqueued {
+    my $lock = shift;
+    !!($enqueued->grep(sub { $_ eq $lock })->size == 1);
+}
+
+sub dequeue {
+    my $lock = shift;
+    $enqueued = $enqueued->grep(sub { $_ ne $lock });
+}
+
+sub enqueue {
+    push @$enqueued, shift;
+}
+
+sub _setup_workers {
+    return @_ unless grep { /worker/i } @_;
+
+    require OpenQA::Worker::Common;
+    my @args = @_;
+    my ($worker_settings, undef) = OpenQA::Worker::Common::read_worker_config(undef, undef);
+    my $cache_workers
+      = exists $worker_settings->{CACHEWORKERS} ? $worker_settings->{CACHEWORKERS} : DEFAULT_MINION_WORKERS;
+    push(@args, '-j', $cache_workers);
+    return @args;
+}
+
+sub run {
+    shift;
+    require OpenQA::Utils;
+    app->log->short(1);
+    OpenQA::Utils::set_listen_address(7844);
+    $ENV{MOJO_INACTIVITY_TIMEOUT} = 300;
```

Comment:
Are you sure you want this hardcoded? Other options would be to use `//=` here or set `$server->inactivity_timeout(300)` in your `before_server_start` hook.

Reply:
Done in https://github.com/os-autoinst/openQA/pull/1783/commits/71e142c959efcf6c2e395f891e7522869be78d7d